### ISSUE TITLE
Connection Lost return code -6

### ISSUE THREAD

**Issue by rosepinkr** [eclipse-paho/paho.mqtt.c#495]
Scenario:
I have subscribed with( /#) MQTT broker through my application.
As soon as I try to publish on the same MQTT broker through my application I lose the subscribe connection from the broker?
With Connection lost, Reconnecting, Failed to start connect, return code -6 ?
Do you know the reason?

**Comment by rosepinkr** [comment]
[traceout.txt](https://github.com/eclipse/paho.mqtt.c/files/2151723/traceout.txt)



I have added the traces.

Now I tried two subscribes one after the other using the MQTTAsync Example to rule out anything happening due to my appilcation. One of the the subscribers looses the connection when the other subscriber joins. 
Kindly let me know where the problem lies. Return Code and Error are same. I am using SSL with test mosquitto.
